contracts/minter-suite/Minters/MinterPolyptych/MinterPolyptychV0.sol

```diff
@@ -0,0 +1,795 @@
+// SPDX-License-Identifier: LGPL-3.0-only
+// Created By: Art Blocks Inc.
+
+import "../../../interfaces/0.8.x/IRandomizerV2.sol";
+import "../../../interfaces/0.8.x/IGenArt721CoreContractV3.sol";
+import "../../../interfaces/0.8.x/IMinterFilterV0.sol";
+import "../../../interfaces/0.8.x/IFilteredMinterHolderV1.sol";
+import "../../../interfaces/0.8.x/IBasicPolyptychRandomizerV0.sol";
+import "../../../interfaces/0.8.x/IDelegationRegistry.sol";
+
+import "@openzeppelin-4.5/contracts/token/ERC721/IERC721.sol";
+import "@openzeppelin-4.5/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin-4.5/contracts/security/ReentrancyGuard.sol";
+import "@openzeppelin-4.5/contracts/utils/structs/EnumerableSet.sol";
+
+pragma solidity 0.8.17;
+
+interface IGenArt721CoreContractV3WithRandomizer is IGenArt721CoreContractV3 {
+    /// current randomizer contract
+    function randomizerContract()
+        external
+        returns (IBasicPolyptychRandomizerV0);
+
+    /**
+     * @notice Returns token hash **seed** for token ID `_tokenId`.
+     * @param _tokenId Token ID to be queried.
+     * @return bytes12 Token hash seed.
+     * @dev token hash seed is keccak256 hashed to give the token hash
+     */
+    function tokenIdToHashSeed(uint256 _tokenId)
+        external
+        view
+        returns (bytes12);
+}
+
+/**
+ * @title Filtered Minter contract that allows tokens to be minted with ETH
+ * when purchaser owns an allowlisted ERC-721 NFT. This contract does NOT track
+ * if a purchaser has/has not minted already -- it simply restricts purchasing
+ * to anybody that holds one or more of a specified list of ERC-721 NFTs.
+ * This is designed to be used with IGenArt721CoreContractV3 contracts.
```

Comment:
I think this should be updated to say it may only be used with V3 Engine contracts that have an active PolyptychRandomizer, and that it is up to admin to appropriately configure the core contract.